cmd/ctl/pkg/status/certificate/certificate.go

```diff
@@ -150,21 +151,88 @@ func (o *Options) Run(args []string) error {
 	fmt.Fprintf(o.Out, fmt.Sprintf("Not After: %s\n", formatTimeString(crt.Status.NotAfter)))
 	fmt.Fprintf(o.Out, fmt.Sprintf("Renewal Time: %s\n", formatTimeString(crt.Status.RenewalTime)))
 
+	// TODO: What about timing issues? When I query condition it's not ready yet, but then looking for cr it's finished and deleted
+	// Try find the CertificateRequest that is owned by crt and has the correct revision
+	reqs, err := o.CMClient.CertmanagerV1alpha2().CertificateRequests(o.Namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	req, err := findMatchingCR(reqs, crt)
+	if err != nil {
+		return err
+	}
+	if req == nil {
+		fmt.Fprintf(o.Out, fmt.Sprintf("No CertificateRequest found for this Certificate\n"))
+	} else {
+		fmt.Fprintf(o.Out, fmt.Sprintf("CertificateRequest:%s\n", crInfoString(req)))
+	}
+
 	// TODO: print information about secret
 	return nil
 }
 
+// formatStringSlice takes in a string slice and formats the contents of the slice
+// into a single string where each element of the slice is prefixed with "- " and on a new line
 func formatStringSlice(strings []string) string {
 	result := ""
-	for _, string := range strings {
-		result += "- " + string + "\n"
+	for _, str := range strings {
+		result += "- " + str + "\n"
 	}
 	return result
 }
 
+// formatTimeString returns the time as a string
+// If nil, return "<none>"
 func formatTimeString(t *metav1.Time) string {
 	if t == nil {
 		return "<none>"
 	}
 	return t.Time.Format(time.RFC3339)
 }
+
+// findMatchingCR tries to find a CertificateRequest that is owned by crt and has the correct revision annotated from reqs.
+// If none found returns nil
+// If one found returns the CR
+// If multiple found returns error
+func findMatchingCR(reqs *cmapi.CertificateRequestList, crt *cmapi.Certificate) (*cmapi.CertificateRequest, error) {
+	possibleMatches := []*cmapi.CertificateRequest{}
+
+	// CertificateRequest revisions begin from 1.
+	// If no revision is set on the Certificate then assume the revision on the CertificateRequest should be 1.
+	// If revision is set on the Certificate then revision on the CertificateRequest should be crt.Status.Revision + 1.
```

Comment:
I seem to remember a conversation about refactoring / sharing some similar existing code that does this same matching, but which uses a `lister`.
If so, add a TODO comment here with a note about the existing code which could be refactored in future.

Reply:
There were two parts to that discussion. One was about sharing the code that's been written in the `predicate` package, which we are doing now after having moved the `predicate` package to pkg/util. I think that's the part you mean.

The other part was about if we should use `lister`s to list resources. That part would not save any code (would be a lot more code I think), and we agreed to just use cert-manager clients.